src/core/Signature.ts

```diff
@@ -0,0 +1,146 @@
+import { RequestBuilder } from '..';
+import { ISignature, Holder, SignatureState, PostSignatureDTO, PatchSignatureDTO } from '../lib';
+
+/**
+ * Signature instance
+ * Cf: https://developers.algoan.com/api/#operation/getSignatureDocument
+ */
+export class Signature implements ISignature {
+  /**
+   * Define when Algoan needs a redirectUrl. It is used to redirect the user after an external process.
+   */
+  public callbackUrl?: string;
+
+  /**
+   * The date of creation of the signature
+   */
+  public createdAt: Date;
+
+  /**
+   * Signatory
+   */
+  public holder: Holder;
+
+  /**
+   * Unique signature document identifier
+   */
+  public id: string;
+
+  /**
+   * An array of legal documents id (signed or not)
+   */
+  public legalDocumentIds: string[];
+
+  /**
+   * metadata about the document to sign (e.g., position of the signature, or information
+   * about the signature success)
+   */
+  public metadata?: unknown;
+
+  /**
+   * Signature id from the Signature Partner
+   */
+  public partnerId: string;
+
+  /**
+   * In case of an electronic signature, redirection link to the provider
+   */
+  public redirectUrl?: string;
+
+  /**
+   * The date of creation of the redirectUrl
+   */
+  public redirectUrlCreatedAt?: Date;
+
+  /**
+   * The max time of validity of the redirectUrl (in second)
+   */
+  public redirectUrlTTL?: number;
+
+  /**
+   * Status of the signature document
+   */
+  public state: SignatureState;
+
+  /**
+   * Folder ID attached to the signature
+   */
+  protected folderId: string;
+
+  constructor(folderId: string, params: ISignature, private readonly requestBuilder: RequestBuilder) {
+    this.folderId = folderId;
+    this.callbackUrl = params.callbackUrl;
+    this.createdAt = new Date(params.createdAt);
```

Comment:
You use a constructor `Date` here. Does it mean `createdAt` is a `string` instead of `Date`? And should I do the same everywhere?

Reply:
The API returns an ISO string, but I think converting it into a date can be useful if the SDK user wants to do operations

Reply:
So, everywhere a `Date` is expected, I should invoke this constructor, right?

Reply:
If the result is an ISO string, yes it could be great

Reply:
Ok, I will change that.